lib/connection.js

```diff
@@ -20,6 +20,8 @@ class Connection extends EventEmitter {
     this.buffer = [];
     this.requestBuffer = [];
     this.requested = 0;
+    // Map <string, function> to register callbacks if responses can be received in any order
+    this.requestCallbacks = new Map();
```

Comment:
Do we need to clear this registry if the connection is closed?

Reply:
Sounds good, added. We have the following code in `close`:

```js
if (this.requestBuffer.length > 0) {
  this.connect();
}
```

And then we are trying to send all requests in `flushRequestBuffer` so it would be helpful to clear registry